src/jsdoc_transformer.ts

```diff
@@ -138,59 +123,55 @@ export function maybeAddHeritageClauses(
       if (!type.symbol) {
         // It's not clear when this can happen.
         mtt.debugWarn(decl, `could not get type of symbol: ${expr.getText()}`);
-        return null;
+        return;
       }
       sym = type.symbol;
     }
+
     if (sym.flags & ts.SymbolFlags.Alias) {
       sym = mtt.typeChecker.getAliasedSymbol(sym);
     }
 
     const typeTranslator = mtt.newTypeTranslator(expr.expression);
     if (typeTranslator.isBlackListed(sym)) {
       // Don't emit references to blacklisted types.
-      return null;
+      return;
     }
 
+    // typeToClosure includes nullability modifiers, so call symbolToString directly here.
+    const parentName = typeTranslator.symbolToString(sym);
+    if (!parentName) return;
+
     if (sym.flags & ts.SymbolFlags.Class) {
       if (!isClass) {
         // Closure interfaces cannot extend or implements classes.
         mtt.debugWarn(decl, `omitting interface deriving from class: ${expr.getText()}`);
-        return null;
+        return;
```

Comment:
Do you intentionally not add to `relatedSuperTypes` in this case? Document why so?

Reply:
To answer this question and the rest, I was attempting to not change the rules about which things were related to which in this change, but just the way we mark the relationships we already were capturing.  But it still fails in google3, which I have finally tracked down, so I'll take another shot at this once I see why it still did the wrong thing.